File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let map = new Map();

        for (let i = 0; i < nums.length; i++) {
            const searchVal = target - nums[i];

            if (map.has(searchVal)) {
                return [map.get(searchVal), i];
            }
            map.set(nums[i], i);
        }
        return [];
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        L, R = 0, len(nums)

        while L < R:
            
            mid = L + (R - L) // 2

            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                L = mid + 1
            else:
                R = mid
            
        return -1
